src/providers/dataProvider/index.test.ts

```diff
@@ -104,3 +115,69 @@ describe('generateBatchId', () => {
     })
   })
 })
+
+describe('generateBatchId for values greater than 9', () => {
+  let provider: DataProvider
+
+  beforeAll(async () => {
+    provider = await getDataProvider()
+    for (let i = 0; i < 20; i++) {
+      await generateBatch(
+        i,
+        provider,
+        year,
+        await generateBatchId(provider, year)
+      )
+    }
+  })
+
+  describe('validate batch numbers', () => {
+    it('should validate first 20 batchNumber and delete first 10 entries', async () => {
+      const batchData = await provider.getList(constants.R_BATCHES, {
+        sort: { field: 'id', order: 'ASC' },
+        pagination: { page: 1, perPage: 1000 },
+        filter: { yearOfReceipt: year }
+      })
+      for (let i = 0; i < batchData.data.length; i++) {
+        expect(batchData.data[i].batchNumber).toBe(
+          `V${i.toLocaleString('en-US', {
+            minimumIntegerDigits: 2,
+            useGrouping: false
+          })}/${year}`
+        )
+      }
+
+      // deleting first 10 entries
+      await provider.delete(constants.R_BATCHES, { id: 10 })
+    })
```

Comment:
The above code looks like it is deleting batch with `id` of 10, rather than the first 10 entries.

The important part of the test comes next though. When we generate a new batch ID, we should see that it provides the value of 20 (max id) rather than 10 (current length)

Reply:
The length is being passed as id due to typescript as delete in provider has this type of arguments. But it is working ok in the mock provider delete function. The next test is generating new 10 batches and then verifying if the new batches have proper ids i.e `V20` on-wards.

Reply:
Aah, I see the issue.  In that case, you should use `deleteMany`, and provide a list of ids to delete.